linkerd/app/outbound/src/http/endpoint.rs

```diff
@@ -54,16 +54,15 @@ where
             }
         }))
         .check_new::<Endpoint>()
-        .push(tap_layer.clone())
+        .push(tap_layer)
         .push(metrics.http_endpoint.into_layer::<classify::Response>())
         .push_on_response(TraceContext::layer(
-            span_sink
-                .clone()
-                .map(|sink| SpanConverter::client(sink, crate::trace_labels())),
+            span_sink.map(|sink| SpanConverter::client(sink, crate::trace_labels())),
         ))
         .push_on_response(http::strip_header::request::layer(L5D_REQUIRE_ID))
         .push(svc::layer::mk(NewRequireIdentity::new))
         .push(http::override_authority::Layer::new(vec![
+            #[allow(clippy::borrow_interior_mutable_const)]
```

Comment:
in general, when we add `allow` attributes that aren't immediately obvious (like `too_many_arguments`), i think it's good to add a comment explaining why we're allowing that lint — here, in particular, i don't totally understand what's going on.